lib/index.js

```diff
@@ -71,68 +55,69 @@ api.validate = (input, validatorConfig, options, callback) => {
-      const publicKeys = [].concat(owner.publicKey);
-      const trustedKeys = publicKeys.filter(
-        k => typeof trustedSigners[k] === 'boolean');
+  let result;
+  try {
+    result = await jsigs.verify(input, {
+      checkKeyOwner: (owner, key, options, callback) => {
+        // owner.publicKey can be a string or an array
+        const publicKeys = [].concat(owner.publicKey);
+        const trustedKeys = publicKeys.filter(
+          k => typeof trustedSigners[k] === 'boolean');
 
-      if(trustedSigners[owner.id] === undefined && trustedKeys.length === 0) {
-        return callback(null, false);
-      }
-      keyOwners[key.id] = owner.id;
-      callback(null, true);
-    }
-  }, (err, result) => {
-    if(err) {
-      return callback(new BedrockError(
-        'An error occurred during signature verification.',
-        'ValidationError', {
-          httpStatusCode: 400,
-          public: true
-        }, err.toString()));
-    }
-    const keyResults = result.keyResults;
-    // trustedSigner may be a publicKey or an owner
-    keyResults.filter(k => k.verified).forEach(k => {
-      if(trustedSigners[k.publicKey] === false) {
-        trustedSigners[k.publicKey] = true;
-      } else if(trustedSigners[keyOwners[k.publicKey]] === false) {
-        trustedSigners[keyOwners[k.publicKey]] = true;
+        if(trustedSigners[owner.id] === undefined && trustedKeys.length === 0) {
+          return callback(null, false);
+        }
+        keyOwners[key.id] = owner.id;
+        callback(null, true);
       }
     });
-
-    let verifiedSignatures = 0;
-    Object.keys(trustedSigners).forEach(
-      s => trustedSigners[s] ? verifiedSignatures++ : null);
-    if(verifiedSignatures < validatorConfig.minimumSignaturesRequired) {
-      return callback(new BedrockError(
-        'The signature requirements have not been met.',
-        'ValidationError', {
-          httpStatusCode: 400,
-          public: true,
-          input,
-          trustedSigners,
-          signatureCount,
-          verifiedSignatures,
-          keyResults,
-          minimumSignaturesRequired: validatorConfig.minimumSignaturesRequired
-        }));
+  } catch(err) {
+    throw new BedrockError(
+      'An error occurred during signature verification.',
+      'ValidationError', {
+        httpStatusCode: 400,
+        public: true
+      }, err.toString());
+  }
+  const keyResults = result.keyResults;
+  // trustedSigner may be a publicKey or an owner
+  keyResults.filter(k => k.verified).forEach(k => {
+    if(trustedSigners[k.publicKey] === false) {
+      trustedSigners[k.publicKey] = true;
+    } else if(trustedSigners[keyOwners[k.publicKey]] === false) {
+      trustedSigners[keyOwners[k.publicKey]] = true;
     }
-    // success
-    callback(null, true);
   });
-};
+
+  let verifiedSignatures = 0;
+  Object.keys(trustedSigners).forEach(
+    s => trustedSigners[s] ? verifiedSignatures++ : null);
```

Comment:
Seems like `Object.values()` and `reduce` would make more sense here, but that's not really part of this PR :)